tests/unit/test.program.js

```diff
@@ -430,6 +432,190 @@ describe('program.main', () => {
       assert.strictEqual(options.shouldExitProgram, false);
     });
   });
+
+  it('applies options from the specified config file', () => {
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+    const fakePath = 'path/to/web-ext-config.js';
+    const resolvedFakePath = path.resolve(fakePath);
+    const fakeLoadJSConfigFile = sinon.spy(() => {});
+    const fakeApplyConfigToArgv = sinon.spy(() => {});
+
+    return execProgram(
+      ['lint', '--config', fakePath],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          applyConfigToArgv: fakeApplyConfigToArgv,
+          loadJSConfigFile: fakeLoadJSConfigFile,
+        },
+      })
+      .then(() => {
+        const options = fakeCommands.lint.firstCall.args[0];
+        assert.strictEqual(options.config, fakePath);
+        sinon.assert.calledOnce(fakeLoadJSConfigFile);
+        sinon.assert.calledWith(fakeLoadJSConfigFile,
+                                sinon.match(resolvedFakePath));
+        sinon.assert.calledOnce(fakeApplyConfigToArgv);
+        sinon.assert.calledWith(fakeApplyConfigToArgv, sinon.match({
+          configFileName: resolvedFakePath,
+        }));
+      });
+  });
+
+  it('throws a UsageError when the config file can\'t be loaded', () => {
+    const fakeProcess = createFakeProcess();
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+    const fakePath = 'path/to/web-ext-config.js';
+    const fakeLoadJSConfigFile = sinon.spy(() => {
+      throw new UsageError('bad path');
+    });
+    const fakeApplyConfigToArgv = sinon.spy(() => {});
+    const capturingConsoleStream = new ConsoleStream();
```

Comment:
@saintsebastian The `ConsoleStream` instance used by the loggers created using the `createLogger` helper function is also exported by the `"util/logger"` module (as `consoleStream`: https://github.com/mozilla/web-ext/blob/master/src/util/logger.js#L94).

Here is a diff applied to this test to make it able to use the default logger instance to make the assertions on the logged messages:

```diff
diff --git a/tests/unit/test.program.js b/tests/unit/test.program.js
index ac73535..ee1a32e 100644
--- a/tests/unit/test.program.js
+++ b/tests/unit/test.program.js
@@ -19,7 +19,10 @@ import {
   makeSureItFails,
   ErrorWithCode,
 } from './helpers';
-import {createLogger, ConsoleStream} from '../../src/util/logger';
+import {
+  createLogger,
+  consoleStream, ConsoleStream,
+} from '../../src/util/logger';

 describe('program.Program', () => {

@@ -474,11 +477,11 @@ describe('program.main', () => {
       throw new UsageError('bad path');
     });
     const fakeApplyConfigToArgv = sinon.spy(() => {});
-    const capturingConsoleStream = new ConsoleStream();
-    capturingConsoleStream.startCapturing();
-    const logger = createLogger(__filename, {
-      logStream: capturingConsoleStream,
-    });
+
+    // Empty any existent capturedMessages.
+    consoleStream.stopCapturing();
+    // Start collecting capturedMessages.
+    consoleStream.startCapturing();

     return execProgram(
       ['lint', '--config', fakePath],
@@ -489,13 +492,16 @@ describe('program.main', () => {
           systemProcess: fakeProcess,
           applyConfigToArgv: fakeApplyConfigToArgv,
           loadJSConfigFile: fakeLoadJSConfigFile,
-          logger,
         },
       })
       .then(makeSureItFails())
       .catch(onlyInstancesOf(UsageError, (error) => {
+        const {capturedMessages} = consoleStream;
+        // empty any existent capturedMessages and stop capturing further logs.
+        consoleStream.stopCapturing();
+
         assert.match(error.message, /bad path/);
-        assert.match(capturingConsoleStream.capturedMessages[0], /bad path/);
+        assert.match(capturedMessages[0], /bad path/);
         sinon.assert.notCalled(fakeApplyConfigToArgv);
         sinon.assert.notCalled(fakeProcess.exit);
```

Reply:
ah! i missed this somehow